mypy_django_plugin/transformers/manytomany.py

```diff
@@ -151,3 +151,52 @@ def get_model_from_expression(
         if model_info is not None:
             return Instance(model_info, [])
     return None
+
+
+def refine_many_to_many_related_manager(ctx: MethodContext) -> MypyType:
+    """
+    Updates the 'ManyRelatedManager' returned by e.g. 'ManyToManyDescriptor' to be a subclass
+    of 'ManyRelatedManager' and the related model's default manager.
+    """
+    if (
+        not isinstance(ctx.default_return_type, Instance)
+        # Only change a return type being 'ManyRelatedManager'
+        or ctx.default_return_type.type.fullname != fullnames.MANY_RELATED_MANAGER
+    ):
+        return ctx.default_return_type
+
+    # This is a call to '__get__' overload with a model instance of 'ManyToManyDescriptor'.
+    # Returning a 'ManyRelatedManager'. Which we want to, just like Django, build from the
+    # default manager of the related model.
+    many_related_manager = ctx.default_return_type
+    # Require first type argument of 'ManyRelatedManager' to be a model
+    if (
+        not many_related_manager.args
+        or not isinstance(many_related_manager.args[0], Instance)
+        or many_related_manager.args[0].type.metaclass_type is None
+        or many_related_manager.args[0].type.metaclass_type.type.fullname != fullnames.MODEL_METACLASS_FULLNAME
```

Comment:
Nit: for some reason I find it difficult to reason about these conditions because they use "inverted logic"

Would make more sense to me as `if args and isinstance(..., Instance) and ... is not None and ...fullname == MODEL_METACLASS_FULLNAME`

Possibly you could refactor these into another function e.g. `related_model_instance = get_related_model_instance(ctx)` that returns `Optional[Type]` and avoid the "inverted logic", without nesting the code too deep. But I'll leave it up to you to decide whether it's worth it.

Could also reduce repetition of `return ctx.default_return_type` a little.